source/extensions/filters/network/echo/echo.h

```diff
@@ -5,12 +5,14 @@
 #include "common/common/logger.h"
 
 namespace Envoy {
-namespace Filter {
+namespace Extensions {
```

Comment:
Why is there an echo.h at all?  Is it included by any other file besides echo.cc?  I guess if it had a test it might be...maybe if it's an example it should have a test even if it's simple?  WDYT?

Thinking about tests, what happens if I run 'bazel test //test/...' without doing whatever bazel flag enables the extension?  I think I'd want that to run the test for the extension anyway.

Along those lines, I was wondering if we could separate out the black magic that auto-registers the filter from the filter implementation.  The unit test wouldn't need the black magic or the build flag, but would initialize it programatically, and it would always run.  The binary would only load the extension with the flag.

Is that conditional loading represented here?

Reply:
In this case, echo.h is not needed as there are no unit tests. It's covered by an integration test. I can remove echo.h if desired. I don't really want to add a unit test just for this change.

> Thinking about tests, what happens if I run 'bazel test //test/...' without doing whatever bazel flag enables the extension? I think I'd want that to run the test for the extension anyway.

This will just work. The integration test correctly includes the filter.

> Along those lines, I was wondering if we could separate out the black magic that auto-registers the filter from the filter implementation. The unit test wouldn't need the black magic or the build flag, but would initialize it programatically, and it would always run. The binary would only load the extension with the flag.

This is the way it works for other filters that have unit tests. There are no changes needed IMO.

> Is that conditional loading represented here?

No. I'm not planning on doing conditional loading as part of this change. I set the stage for it in all_extensions.bzl which we can flesh out later IMO.